Cleanup for object manager references and depricated method

atishgoswami opened the pull request:
Contact Module Cleanups

## Description

This PR will:

- Removes object manager references
- Remove deprecated call to $messageManager
- Changed `frontend` area string to `Magento\Framework\App\Area::AREA_FRONTEND` constant

## Fixed Issues (if relevant)

None, code improvement

## Contribution checklist

- [x] Pull request has a meaningful description of its purpose
- [x] All commits are accompanied by meaningful commit messages
- [x] All new or changed code is covered with unit/integration tests (if applicable)
- [x] All automated tests passed successfully (all builds on Travis CI are green)

atishgoswami committed:
Cleanup for object manager references and depricated method

ihor-sviziev commented on app/code/Magento/Contact/Controller/Index/Post.php:
```diff
@@ -53,10 +53,10 @@ public function __construct(
+        $this->mail          = $mail;
         $this->dataPersistor = $dataPersistor;
-        $this->logger = $logger ?: \Magento\Framework\App\ObjectManager::getInstance()->get(LoggerInterface::class);
+        $this->logger        = $logger;
```
This change is not backward compatible, so please revert it.

ihor-sviziev commented on app/code/Magento/Contact/Model/Mail.php:
```diff
@@ -44,20 +44,19 @@ public function __construct(
         $this->inlineTranslation = $inlineTranslation;
-        $this->storeManager = $storeManager ?:
-            ObjectManager::getInstance()->get(StoreManagerInterface::class);
+        $this->storeManager      = $storeManager;
```
This change is not backward compatible, so please revert it.

ihor-sviziev commented on app/code/Magento/Contact/Model/Mail.php:
```diff
@@ -44,20 +44,19 @@ public function __construct(
         TransportBuilder $transportBuilder,
         StateInterface $inlineTranslation,
-        StoreManagerInterface $storeManager = null
+        StoreManagerInterface $storeManager
```
This change is not backward compatible, so please revert it.

ihor-sviziev reviewed (commented):
1. We can't remove optional parameters in constructor due to backward compatibility
2. Replacing addSuccess to addSuccessMessage - looks good
3. Remove getDataPersistor and use property instead - looks good
4. Re-formatting code just to reformat - I think it looked better
5. Using constant for frontend area - good

atishgoswami committed:
Reverted backward incompatible changes

atishgoswami commented:
Hello @ihor-sviziev,
As suggested removed all backward incompatible changes. Please review and let me know if any further changes required.

ihor-sviziev commented:
@atishgoswami integration tests failed. Could you check it?

atishgoswami commented:
@ihor-sviziev Yes working on it

ishakhsuvarov committed:
magento/magento2#12061: Merge branch '2.2-develop' of github.com:magento/magento2 into contact-module-cleanups

ishakhsuvarov committed:
magento/magento2#12061: Cleanup for object manager references and deprecated method
 - Updated coding style to match the one used in core magento
 - Updated integration test to verify for the escaped string

ihor-sviziev commented on dev/tests/integration/testsuite/Magento/Contact/Controller/IndexTest.php:
```diff
@@ -24,7 +24,7 @@ public function testPostAction()
         $this->assertSessionMessages(
             $this->contains(
-                "Thanks for contacting us with your comments and questions. We'll respond to you very soon."
+                "Thanks for contacting us with your comments and questions. We&#039;ll respond to you very soon."
```
Not sure that it will be good idea to make it escaped there. Have you checked this change?

ishakhsuvarov replied:
Yes, I have checked this manually. On Contact Form It works as expected:
<img width="616" alt="contact_form_html_entities" src="https://user-images.githubusercontent.com/5747062/32606640-b50fa236-c55e-11e7-8579-e3f824dd3cbd.png">

Since this test is testing the controller I assume that it is valid to check for the exact html we expect.

atishgoswami commented:
@ishakhsuvarov Thanks for fixing the tests 👍
